src/meta-srv/src/procedure/region_migration.rs

```diff
@@ -118,15 +126,47 @@ pub struct PersistentContext {
     pub(crate) trigger_reason: RegionMigrationTriggerReason,
 }
 
+impl PersistentContext {
+    pub fn new(
+        catalog_and_schema: Vec<(String, String)>,
+        from_peer: Peer,
+        to_peer: Peer,
+        region_ids: Vec<RegionId>,
+        timeout: Duration,
+        trigger_reason: RegionMigrationTriggerReason,
+    ) -> Self {
+        #[allow(deprecated)]
+        Self {
+            catalog: None,
+            schema: None,
+            catalog_and_schema,
+            from_peer,
+            to_peer,
+            region_ids,
+            timeout,
+            trigger_reason,
+        }
+    }
+}
+
 fn default_timeout() -> Duration {
     Duration::from_secs(10)
 }
 
 impl PersistentContext {
     pub fn lock_key(&self) -> Vec<StringKey> {
-        let mut lock_keys = Vec::with_capacity(self.region_ids.len() + 2);
-        lock_keys.push(CatalogLock::Read(&self.catalog).into());
-        lock_keys.push(SchemaLock::read(&self.catalog, &self.schema).into());
+        let mut lock_keys =
+            Vec::with_capacity(self.region_ids.len() + 2 + self.catalog_and_schema.len() * 2);
+        #[allow(deprecated)]
+        if let (Some(catalog), Some(schema)) = (&self.catalog, &self.schema) {
+            lock_keys.push(CatalogLock::Read(catalog).into());
+            lock_keys.push(SchemaLock::read(catalog, schema).into());
+        }
+        for (catalog, schema) in self.catalog_and_schema.iter() {
+            lock_keys.push(CatalogLock::Read(catalog).into());
+            lock_keys.push(SchemaLock::read(catalog, schema).into());
+        }
```

Comment:
If `catalog_and_schema` is not empty, then there is no need to process `catalog` and `schema`, right?

Reply:
Yes